backend/src/events.py

```diff
@@ -40,6 +40,9 @@ class NodeFinishData(TypedDict):
 
 class IteratorProgressUpdateData(TypedDict):
     percent: float
+    index: int
+    total: int
```

Comment:
You don't use those 2 in the frontend, so why transmit them?

Btw. it's good that you don't use them in the frontend, because you send slightly wrong indexes in a few places.

Reply:
which places do I send the wrong index?